ANMN/burst_averaged_product/burst_average_product.py

```diff
@@ -0,0 +1,388 @@
+        if var_to_remove in var_list:
+            var_list.remove(var_to_remove)
+
+    return var_list
+
+def burst_average_data(time_values, var_values, burst_duration, var_qc_exclusion):
+    """
+    create burst average data for var_values. This is the core of the script
+    """
+    n_seconds_day = 24 * 3600
+    difft         = np.round(np.diff(time_values) * n_seconds_day) # this gives us a spike for each new burst
+    allowed_gap   = burst_duration
+    fill_value    = float('nan')
+
+    # look for start index of each burst, ie the index of each new spike of
+    # difft variable
+    idx_spike = [0] # initialise first index
+    spikes    = np.where(difft >= allowed_gap)[0] + 1
+    idx_spike.extend(spikes)
+
+    # initialise arrays
+    time_mean_burst   = []
+    var_mean_burst    = []
+    var_min_burst     = []
+    var_max_burst     = []
+    var_std_burst     = []
+    var_num_obs_burst = []
+
+    for idx_spike_idx, idx_spike_val in enumerate(idx_spike[:-1]):
+        index_burst_range = range(idx_spike_val, idx_spike[idx_spike_idx + 1])
+
+        # burst average of TIME variable. All the range of the burst is used
+        time_mean_burst.append((time_values[index_burst_range[0]] + time_values[index_burst_range[-1]]) / 2)
+
+        # For NON TIME variables, the operations are only performed in respect
+        # to boolean values found in var_qc_exclusion variable.
+        # this var_qc_exclusion variable is set to True for all IMOS IODE flags
+        # greater than a specific value
+        valid_index_burst_range = [val for val in index_burst_range if not var_qc_exclusion[val]]
+
+        # condition in case no good qc data was found for one burst
+        if valid_index_burst_range:
+            var_mean_burst   .append(np.mean(var_values[valid_index_burst_range]))
+            var_min_burst    .append(np.min(var_values[valid_index_burst_range]))
+            var_max_burst    .append(np.max(var_values[valid_index_burst_range]))
+            var_std_burst    .append(np.std(var_values[valid_index_burst_range]))
+            var_num_obs_burst.append(len(valid_index_burst_range))
+        else:
+            var_mean_burst   .append(fill_value)
+            var_min_burst    .append(fill_value)
+            var_max_burst    .append(fill_value)
+            var_std_burst    .append(fill_value)
+            var_num_obs_burst.append(0)
+
+    return time_mean_burst, var_mean_burst, var_min_burst, var_max_burst, var_std_burst, var_num_obs_burst
+
+def generate_netcdf_burst_filename(netcdf_file_obj, burst_vars):
+    """
+    generate the filename of a burst average netcdf for both CTD and WQM
+    """
+    site_code         = netcdf_file_obj.site_code
+    input_netcdf_name = os.path.basename(input_netcdf_file_path)
+    pattern           = re.compile("^(IMOS_.*)_([0-9]{8}T[0-9]{6}Z)_(.*)_(FV0[0-9])_(.*)_END")
+    match_group       = pattern.match(input_netcdf_name)
+
+    time_burst_vals = burst_vars['TEMP'][0]
```

Comment:
no, because the keys of burst_vars are only variable to average. 
time -> burst_vars['TEMP'][0]
mean_temp -> burst_vars['TEMP'][1]
...